src/monodroid/jni/embedded-assemblies.cc

```diff
@@ -316,63 +301,42 @@ md_mmap_seek_file (void *opaque, void *stream, uLong offset, int origin)
 	return 0;
 }
 
-static int
-md_mmap_close_file (void *opaque, void *stream)
+int
+EmbeddedAssemblies::md_mmap_close_file (UNUSED_ARG void *opaque, void *stream)
 {
 	free (stream);
 	return 0;
 }
 
-static int
-md_mmap_error_file (void *opaque, void *stream)
+int
+EmbeddedAssemblies::md_mmap_error_file (UNUSED_ARG void *opaque, UNUSED_ARG void *stream)
 {
 	return 0;
 }
 
-static mono_bool
-register_debug_symbols_for_assembly (DylibMono *mono, const char *entry_name, MonoBundledAssembly *assembly, const mono_byte *debug_contents, int debug_size)
+bool
+EmbeddedAssemblies::register_debug_symbols_for_assembly (const char *entry_name, MonoBundledAssembly *assembly, const mono_byte *debug_contents, int debug_size)
 {
 	const char *entry_basename  = strrchr (entry_name, '/') + 1;
 	// System.dll, System.dll.mdb case
 	if (strncmp (assembly->name, entry_basename, strlen (assembly->name)) != 0) {
 		// That failed; try for System.dll, System.pdb case
 		const char *eb_ext  = strrchr (entry_basename, '.');
 		if (eb_ext == nullptr)
-			return 0;
+			return false;
 		int basename_len    = eb_ext - entry_basename;
 		assert (basename_len > 0 && "basename must have a length!");
 		if (strncmp (assembly->name, entry_basename, basename_len) != 0)
-			return 0;
+			return false;
 	}
 
-	mono->register_symfile_for_assembly (assembly->name, debug_contents, debug_size);
+	monoFunctions.register_symfile_for_assembly (assembly->name, debug_contents, debug_size);
 
-	return 1;
-}
-
-MONO_API int
-monodroid_embedded_assemblies_set_assemblies_prefix (const char *prefix)
```

Comment:
There is Another Repo that cares about our "public API" surface: https://github.com/mono/Embeddinator-4000

In particular, they call `monodroid_embedded_assemblies_set_assemblies_prefix()`:

https://github.com/mono/Embeddinator-4000/blob/c237e21b63df40799ef251350ee961846c549553/support/mono_embeddinator.c#L62

As such:

 1. This export shouldn't be removed, and
 2. I don't think `EmbeddedAssemblies::assemblies_prefix` can actually be a `constexpr`.

Reply:
1. is it just this API or all of the ones which were here before?
2. it can - it's the default, we can use it as long as the 3rd party sets a new one, in which case we'll use the new value

Reply:
@jonpryor you understand `constexpr` correctly, but see the updated code [here](https://github.com/xamarin/xamarin-android/pull/2606/files#diff-46c984aa2baae361655e0ca3bcf0e856R70). We should optimize for the most common case - and that is our prefix in this instance.

Reply:
`monodroid_embedded_assemblies_set_assemblies_prefix` is the only one called by Embeddinator.

I was not able to put assembly files in `/assemblies`, because AAR files couldn't have arbitrary directories like that. I put them in `assets/assemblies` and called `monodroid_embedded_assemblies_set_assemblies_prefix`. I had to include instructions to leave `.dll` files uncompressed.

Reply:
`monodroid_embedded_assemblies_set_assemblies_prefix` is the only thing Embeddinator uses from `libmonondroid.so`. It has to put .NET assemblies into `assets/assemblies` instead of `assemblies`.